rts/lra/test/arquillian-extension/src/main/java/io/narayana/lra/arquillian/deployment/scenario/LRACoordinatorScenarioGenerator.java

```diff
@@ -0,0 +1,121 @@
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+public class LRACoordinatorScenarioGenerator extends ScenarioGeneratorBase implements DeploymentScenarioGenerator {
+
+    public static final String EXTENSION_NAME = "LRACoordinatorDeployment";
+    public static final String EXTENSION_DEPLOYMENT_METHOD = "deploymentMethod"; // needed
+    public static final String EXTENSION_DEPLOYMENT_NAME = "deploymentName"; // needed
+    public static final String EXTENSION_GROUP_NAME = "groupName"; // needed
+    public static final String EXTENSION_CONTAINER_NAME = "containerName"; // needed
+    public static final String EXTENSION_TESTABLE = "testable";
+
+    @Override
+    public List<DeploymentDescription> generate(TestClass testClass) {
+
+        List<DeploymentDescription> descriptions = new ArrayList<>();
+
+        // Fetch all properties in the section EXTENSION_NAME
+        Map<String, String> extensionProperties = getExtensionProperties(EXTENSION_NAME);
+
+        // If the section of this extension is not in the arquillian.xml file, it means that this extension
+        // does not need to start. As a consequence, an empty list of DeploymentDescription is returned
+        if (extensionProperties == null) {
+            return new ArrayList<>();
+        }
+
+        // Checks that all required properties are defined
+        checkPropertiesExistence(
+                extensionProperties,
+                Arrays.asList(
+                        EXTENSION_DEPLOYMENT_METHOD,
+                        EXTENSION_DEPLOYMENT_NAME,
+                        EXTENSION_GROUP_NAME,
+                        EXTENSION_CONTAINER_NAME,
+                        EXTENSION_TESTABLE));
+
+        try {
+            Method deploymentMethod = getDeploymentMethodFromConfiguration(extensionProperties, EXTENSION_DEPLOYMENT_METHOD);
+            GroupDef group = getGroupWithName(extensionProperties.getOrDefault(EXTENSION_GROUP_NAME, ""));
+
+            ContainerDef container;
+            if (group != null) {
+                container = getContainerWithName(group, extensionProperties.get(EXTENSION_CONTAINER_NAME));
+            } else {
+                container = getContainerWithName(extensionProperties.get(EXTENSION_CONTAINER_NAME));
+                if (container == null) {
+                    String message = String.format(
+                            "%s: no container was found with name: %s.",
+                            EXTENSION_NAME,
+                            extensionProperties.get(EXTENSION_CONTAINER_NAME));
+
+                    log.error(message);
+                    throw new RuntimeException(message);
+                }
+            }
+
+            String containerName = container.getContainerName();
+
+            // Instantiates an object of the specified class (using the default constructor)
+            Object obj = deploymentMethod.getDeclaringClass().getDeclaredConstructor().newInstance(null);
```

Comment:
Reflection...

Reply:
I think that reflection makes the overall logic of this class more generic. In other words, when using the `arquillian-extension` module, all parameters will be passed through the `arquillian.xml` where `arquillian-extension` is actually used (e.g. `rts/lra/test/basic`). Of course, if you and @xstefank think that reflection is deal breaker here, I can simplify this class (e.g. hard-coding the deployment function) and scrap entirely the reflection part. As one of the consequences, we will have pre-defined names for Arquillian groups and containers.